src/internal/observable/combineLatest.ts

```diff
@@ -510,87 +510,108 @@ export function combineLatest<O extends ObservableInput<any>, R>(
 
   const { args: observables, keys } = argsArgArrayOrObject(args);
 
-  return lift(fromArray(observables, scheduler), new CombineLatestOperator(resultSelector, keys));
-}
-
-export class CombineLatestOperator<T, R> implements Operator<T, R> {
-  constructor(private resultSelector: ((...values: Array<any>) => R) | undefined, private keys: Array<string> | null) {}
+  const result = new Observable<ObservedValueOf<O>[]>(
+    combineLatestInit(
+      observables as ObservableInput<ObservedValueOf<O>>[],
+      scheduler,
+      keys
+        ? // A handler for scrubbing the array of args into a dictionary.
+          (args: any[]) => {
+            const value: any = {};
+            for (let i = 0; i < args.length; i++) {
+              value[keys![i]] = args[i];
+            }
+            return value;
+          }
```

Comment:
question: Will something like this be needed for `forkJoin`, too? Should it be extracted into another args-related util?

Reply:
Probably. I figured I'd cross the bridge when I came to it (after this is merged)